src/cr/cube/cube_slice.py

```diff
@@ -26,134 +54,88 @@ def _update_args(self, kwargs):
-
-    @lazyproperty
-    def dimensions(self):
-        '''Get slice dimensions.
-
-        For 2D cubes just get their dimensions. For 3D cubes, don't get the
-        first dimension (only take slice dimensions).
-        '''
-        return self._cube.dimensions[-2:]
-
-    # API Methods
-
-    def labels(self, *args, **kwargs):
-        '''Return correct labels for slice.'''
-        return self._cube.labels(*args, **kwargs)[-2:]
-
-    def prune_indices(self, *args, **kwargs):
-        '''Extract correct row/col prune indices from 3D cube.'''
-        if self.ndim < 3:
-            return self._cube.prune_indices(*args, **kwargs)
-        return list(self._cube.prune_indices(*args, **kwargs)[self._index])
-
-    def as_array(self, *args, **kwargs):
-        '''Call cube's as_array, and return correct slice.'''
-        return self._call_cube_method('as_array', *args, **kwargs)
-
-    def proportions(self, *args, **kwargs):
-        '''Call cube's proportions, and return correct slice.'''
-        return self._call_cube_method('proportions', *args, **kwargs)
-
-    def margin(self, *args, **kwargs):
-        '''Call cube's margin, and return correct slice.'''
-        return self._call_cube_method('margin', *args, **kwargs)
-
-    def population_counts(self, *args, **kwargs):
-        '''Get population counts.'''
-        return self._call_cube_method('population_counts', *args, **kwargs)
-
-    @lazyproperty
-    def standardized_residuals(self):
-        '''Get cube's standardized residuals.'''
-        return self._cube.standardized_residuals
-
-    def index(self, *args, **kwargs):
-        '''Get index.'''
-        return self._call_cube_method('index', *args, **kwargs)
-
-    def zscore(self, *args, **kwargs):
-        '''Get index.'''
-        return self._call_cube_method('zscore', *args, **kwargs)
-
-    @lazyproperty
-    def dim_types(self):
-        '''Get dimension types of the cube slice.'''
-        return self._cube.dim_types[-2:]
-
-    @lazyproperty
-    def pvals(self):
-        '''Get pvals of the cube.'''
-        return self._cube.pvals
-
-    def inserted_hs_indices(self, *args, **kwargs):
-        '''Get inserted H&S indices.'''
-        return self._cube.inserted_hs_indices(*args, **kwargs)[-2:]
+    @property
+    def row_dim_ind(self):
+        """
+        Index of the row dimension in the cube
+        :rtype: int
+        """
+        return 0
+
+    @property
+    def col_dim_ind(self):
+        """
+        Index of the column dimension in the cube
+        :rtype: int
+        """
+        return 1
+
+    @property
+    def has_ca(self):
+        return 'categorical_array' in self.dim_types
+
+    @property
+    def mr_dim_ind(self):
+        mr_dim_ind = self._cube.mr_dim_ind
+        if self.ndim == 3:
+            if isinstance(mr_dim_ind, int):
+                return mr_dim_ind - 1
+            elif isinstance(mr_dim_ind, tuple):
+                return tuple([i - 1 for i in mr_dim_ind])
```

Comment:
Remove the square braces. No need to create a list.

Reply:
Done